### PR TITLE
Dedicated RFID field

### PR DESCRIPTION
**Pull request by TomekFraczek**
Added a dedicated RFID field along with associated form field and widget.

The widget is designed to have two states:

- Scan mode:
This state is intended to make it convenient to scan a rfid tag without accidentally editing the number. It has two buttons, 'Scan RFID' which places the focus on a hidden text input, and 'Manual entry' Which switches the widget to manual entry mode.

- Manual entry mode
This mode is intended to make it possible to manually enter the rfid, in case no scanner is available. It has two parts, a text entry field, and a button to switch to scan mode.

Progess:
- [x] Add field that validates RFID (length and only  numbers)
- [x] Add form field
- [x] Include form field on all models

[EDIT]: I've given up on the widget. The code is still there, but I'm having trouble focusing the cursor while keeping everything editable and or hidden as needed. The extra benefit is just not worth the time right now.

### CODE REVIEW

File: core/forms/fields/RFIDField.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from django import forms
+from django.forms import ValidationError
+from ..widgets.RFIDWidget import RFIDWidget
```
Comment (estensen) [inline core/forms/fields/RFIDField.py]: Is it really called `..widgets.RFIDWidget`?

Comment (estensen) [reply core/forms/fields/RFIDField.py]: I mean, why can't you import it from the current dir?

File: core/models/fields/RFIDField.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    """
+    Very simple model field specifically designed to store RFID's.
+
+    Essentially just a thin wrapper around CharField that forces the RFID to be unique and ensures that it has at most
+    10 characters.
+    """
+
+    description = "Specific type of charfield for storing 10 digit RFID's"
+
+    def __init__(self, *args, **kwargs):
```
Comment (estensen) [inline core/models/fields/RFIDField.py]: `*args` and `**kwargs` can be scary for new programmers. Are you sure you need to pass a variable number of arguments to the function?

File: core/models/fields/RFIDField.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        kwargs['max_length'] = 10
+        kwargs['unique'] = True
+        super().__init__(*args, **kwargs)
+
+    def deconstruct(self):
```
Comment (estensen) [inline core/models/fields/RFIDField.py]: unused method

File: core/models/fields/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from .RFIDField import RFIDField
```
[the remark this answers is not in the document]
Comment (TomekFraczek) [reply core/models/fields/__init__.py]: Sure. Any specific reason for that though? Style?

File: core/models/fields/RFIDField.py (hunk 2 above)
Comment (TomekFraczek) [reply core/models/fields/RFIDField.py]: This is the way Django recommends doing it.

https://docs.djangoproject.com/en/2.0/ref/models/fields/

It allows users further down the road to use the options on the underlying Char field, such as default value, validators, etc 

https://docs.djangoproject.com/en/2.0/ref/models/fields/

File: core/models/fields/RFIDField.py (hunk 3 above)
Comment (TomekFraczek) [reply core/models/fields/RFIDField.py]: https://docs.djangoproject.com/en/2.0/howto/custom-model-fields/#WritingASubclass

**Comment by TomekFraczek** [comment]
I'm giving up on the whole widget thing. Too many little issues for it to be worth it.

**Pull request merged**